core/src/main/java/edu/wpi/grip/core/operations/networktables/NTPublishOperation.java

```diff
@@ -0,0 +1,140 @@
+     * which as the hostname that looks like "roborio-frc-xxxx.local".
+     */
+    @Subscribe
+    public void updateSettings(ProjectSettingsChangedEvent event) {
+        final ProjectSettings projectSettings = event.getProjectSettings();
+
+        NetworkTable.shutdown();
+        if (projectSettings.getNetworkProtocol() == ProjectSettings.NetworkProtocol.NETWORK_TABLES) {
+            NetworkTable.setIPAddress("roborio-" + projectSettings.getTeamNumber() + "-frc.local");
+            NetworkTable.initialize();
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "Publish " + type.getSimpleName();
+    }
+
+    @Override
+    public String getDescription() {
+        return "Publish a " + type.getSimpleName() + " to NetworkTables";
+    }
+
+    @Override
+    public Optional<InputStream> getIcon() {
+        return Optional.of(getClass().getResourceAsStream("/edu/wpi/grip/ui/icons/first.png"));
+    }
+
+    @Override
+    public InputSocket<?>[] createInputSockets(EventBus eventBus) {
+        final InputSocket<?>[] sockets = new InputSocket[2 + ntValueMethods.size()];
+        int i = 0;
+
+        // Create an input for the actual object being published
+        sockets[i++] = new InputSocket<>(eventBus,
+                new SocketHint.Builder<>(type).identifier("Value").initialValue(null).build());
+
+        // Create a string input for the key used by NetworkTables
+        sockets[i++] = new InputSocket<>(eventBus,
+                SocketHints.Inputs.createTextSocketHint("Subtable Name", "my" + type.getSimpleName()));
+
+        // Create a checkbox for every property of the object that might be published.  For example, for a
+        // ContourReport, the user might wish to publish the x and y coordinates of the center of each contour.
+        for (Method method : ntValueMethods) {
+            sockets[i++] = new InputSocket<>(eventBus,
+                    SocketHints.createBooleanSocketHint("Publish " + method.getAnnotation(NTValue.class).key(), true));
+        }
+
+        return sockets;
+    }
+
+    @Override
+    public OutputSocket<?>[] createOutputSockets(EventBus eventBus) {
+        return new OutputSocket<?>[0];
+    }
+
+    @Override
+    public void perform(InputSocket<?>[] inputs, OutputSocket<?>[] outputs) {
+        int i = 0;
+
+        final NTPublishable value = (NTPublishable) inputs[i++].getValue().get();
+        final String subtableName = (String) inputs[i++].getValue().get();
+
+        if (subtableName.isEmpty()) {
+            throw new IllegalArgumentException("Need key to publish to NetworkTables");
+        }
+
+        // Get a subtable to put the values in.  Each NTPublishable has multiple properties that are published (such as
+        // x, y, width, height, etc...), so they're grouped together in a subtable.
+        final ITable subtable = table.getSubTable(subtableName);
```

Comment:
I don't know how to do that besides manually keeping a list of what table names we're using, and I don't think it's necessary